ee/scripts/job/run-job.sh

```diff
@@ -69,8 +70,22 @@ copy_artifacts() {
   echo "Artifacts: ($((FILE_COUNT)) files / $((TOTAL_SIZE)) bytes) uploaded"
 }
 
+# Grant sudo
+if [[ "${GRANT_SUDO}" == "true" ]]; then
+  if [[ -n $NB_USER ]]; then
+    USER=$NB_USER
+  fi
+  if [[ -n $USER ]] && [[ -d /etc/sudoers.d ]]; then
+    echo "$USER ALL=(ALL) NOPASSWD:ALL" > /etc/sudoers.d/job
+  fi
+fi
+
 # Run Command
-bash -c "$COMMAND"
+if [[ "$GRANT_SUDO" == "true" ]] && [[ -n $USER ]]; then
+  exec sudo -E -H -u $USER PATH=$PATH bash -c "$COMMAND"
```

Comment:
Should we check if sudo command is available?.

```
if command -v sudo > /dev/null && [[ "$GRANT_SUDO" == "true" ]] && [[ -n $USER ]]; then
```

Reply:
Sounds great